controller/appcontroller.go

```diff
@@ -1771,6 +1771,21 @@ func (ctrl *ApplicationController) needRefreshAppStatus(app *appv1.Application,
 	compareWith := CompareWithLatest
 	refreshType := appv1.RefreshTypeNormal
 
+	if val, ok := app.Annotations[appv1.AnnotationAppRefreshTimeout]; ok {
```

Comment:
This is better to compute on a higher-level function and keep this function just checking things for the specified params. Overriding passed in params is confusing.